test/system/src/main/java/io/pravega/test/system/framework/services/docker/DockerBasedService.java

```diff
@@ -0,0 +1,158 @@
+        }
+        return serviceId;
+    }
+
+    @Override
+    public boolean isRunning() {
+        boolean value = false;
+        try {
+            //list the service with filter 'serviceName'
+            Service.Criteria criteria = Service.Criteria.builder().serviceName(this.serviceName).build();
+            List<Service> serviceList = dockerClient.listServices(criteria);
+
+            for (int i = 0; i < serviceList.size(); i++) {
+                String serviceId = serviceList.get(i).id();
+                if (!serviceId.isEmpty()) {
+                    value = true;
+                    break;
+                }
+            }
+        } catch (DockerException | InterruptedException e) {
+            log.error("Unable to list docker services {}", e);
+        }
+        return value;
+    }
+
+    CompletableFuture<Void> waitUntilServiceRunning() {
+        log.debug("IS RUNNING {}", isRunning());
+        return FutureHelpers.loop(() -> !isRunning(), //condition
+                () -> FutureHelpers.delayedFuture(Duration.ofSeconds(5), executorService),
+                executorService);
+    }
+
+    String setSystemProperty(final String propertyName, final String propertyValue) {
+        return new StringBuilder().append(" -D").append(propertyName).append("=").append(propertyValue).toString();
+    }
+
+    @Override
+    public List<URI> getServiceDetails() {
+        Service.Criteria criteria = Service.Criteria.builder().serviceName(this.serviceName).build();
+        List<URI> uriList = new ArrayList<>();
+        try {
+            List<Service> serviceList = dockerClient.listServices(criteria);
+
+            for (int i = 0; i < serviceList.size(); i++) {
+                URI uri = URI.create("tcp://" + dockerClient.getHost() + ":"
+                        + serviceList.get(i).spec().endpointSpec()
+                        .ports().get(i).targetPort());
+                uriList.add(uri);
+            }
+        }  catch (InterruptedException | DockerException e) {
+            log.error("Unable to list service details {}", e);
+        }
+        return uriList;
+    }
+
+    @Override
+    public void scaleService(final int instanceCount, final boolean wait) {
+        try {
+            Preconditions.checkArgument(instanceCount >= 0, "negative value: %s", instanceCount);
+            Service.Criteria criteria = Service.Criteria.builder().serviceName(this.serviceName).build();
+            String serviceId = dockerClient.listServices(criteria).get(0).id();
+            dockerClient.updateService(serviceId, 1L, ServiceSpec.builder().mode(ServiceMode.withReplicas(instanceCount)).build());
+            String updateState = dockerClient.inspectService(serviceId).updateStatus().state();
+            log.info("update state {}", updateState);
+            if (wait) {
+                waitUntilServiceRunning().get();
+            }
+        } catch (ExecutionException | DockerException | InterruptedException e) {
```

Comment:
I don't think we want to be suppressing exception all over like this. 
Also, do not suppress Interrupted

Reply:
done.